File: hyperledger-fabric/go/helloBlock.go
package main

import (
	"errors"
	"fmt"

	"github.com/hyperledger/fabric-contract-api-go/contractapi"
)

// The start of the helloBlock chaincode implementation
type helloBlock struct {
	contractapi.Contract
}

// Initializing the chaincode so that it starts with a state
func (t *helloBlock) Init(ctx contractapi.TransactionContextInterface, state string) error {
	fmt.Println("helloBlock initialization")
	var err error
	fmt.Printf("State %s", state)
	// Write the initialized state to the ledger
	err = ctx.GetStub().PutState(state, []byte(state))
	if err != nil {
		return err
	}

	return nil
}

// Invoking the chaincode to write a state to the ledger
func (t *helloBlock) Invoke(ctx contractapi.TransactionContextInterface, state string) error {
	var err error
	// Get the state from the ledger
	statebytes, err := ctx.GetStub().GetState(state)
	if err != nil {
		return fmt.Errorf("Cannot get state")
	}
	if statebytes == nil {
		return fmt.Errorf("No state found")
	}

	fmt.Printf("State %s", state)

	// Write the state back to the ledger
	err = ctx.GetStub().PutState(state, []byte(state))
	if err != nil {
		return err
	}

	return nil
}

// Query the state
func (t *helloBlock) Query(ctx contractapi.TransactionContextInterface, state string) (string, error) {
	var err error
	// Get the state from the ledger
	statebytes, err := ctx.GetStub().GetState(state)
	if err != nil {
		jsonResp := "{\"Cannot get state for " + state + "\"}"
		return "", errors.New(jsonResp)
	}

	if statebytes == nil {
		jsonResp := "{\"No state for " + state + "\"}"
		return "", errors.New(jsonResp)
	}

	jsonResp := "{\"" + state + "\",\"" + string(statebytes) + "\"}"
	fmt.Printf("Query response:%s\n", jsonResp)
	return string(statebytes), nil
}

func main() {
	cc, err := contractapi.NewChaincode(new(helloBlock))
	if err != nil {
		panic(err.Error())
	}
	if err := cc.Start(); err != nil {
		fmt.Printf("Error starting helloBlock chaincode: %s", err)
	}
}

File: hyperledger-fabric/javascript/helloBlock.js

const shim = require('fabric-shim');

// The start of the helloBlock chaincode implementation
var helloBlock = class {

  // Initializing the chaincode so that it starts with a state
  async Init(stub) {
    console.info('helloBlock initialization');
    let ret = stub.getFunctionAndParameters();
    console.info(ret);
    let args = ret.params;

    let state = args[0];
    
    // Write the initialized state to the ledger
    try {
      await stub.putState(state, args[0]);
      return shim.success();
    } catch (err) {
      return shim.error(err);
    }
  }

  // Invoking the chaincode to write a state to the ledger
  async Invoke(stub) {
    let ret = stub.getFunctionAndParameters();
    console.info(ret);
    let method = this[ret.fcn];
    if (!method) {
      console.log('No' + ret.fcn + ' command found');
      return shim.success();
    }
    try {
      let payload = await method(stub, ret.params);
      return shim.success(payload);
    } catch (err) {
      console.log(err);
      return shim.error(err);
    }
  }

  async invoke(stub, args) {

    let state = args[0];

    if (!state) {
      throw new Error('State cannot be empty');
    }

    // Get the state from the ledger
    let statebytes = await stub.getState(state);
    if (!statebytes) {
      throw new Error('Cannot get state');
    }

    // Write the state back to the ledger
    await stub.putState(state, args[0]);

  }

  // Query the state
  async query(stub, args) {
    let jsonResp = {};
    let state = args[0];

    // Get the state from the ledger
    let statebytes = await stub.getState(state);
    if (!statebytes) {
      jsonResp.error = 'Cannot get state for ' + state;
      throw new Error(JSON.stringify(jsonResp));
    }

    jsonResp.name = state;
    jsonResp.amount = statebytes.toString();
    console.info('Query response:');
    console.info(jsonResp);
    return statebytes;
  }
};

shim.start(new helloBlock());
